example_11/hardware/carlikebot_system.cpp

```diff
@@ -0,0 +1,315 @@
+}
+
+hardware_interface::CallbackReturn CarlikeBotSystemHardware::on_activate(
+  const rclcpp_lifecycle::State & /*previous_state*/)
+{
+  RCLCPP_INFO(rclcpp::get_logger("CarlikeBotSystemHardware"), "Activating ...please wait...");
+
+  for (auto i = 0; i < hw_start_sec_; i++)
+  {
+    rclcpp::sleep_for(std::chrono::seconds(1));
+    RCLCPP_INFO(
+      rclcpp::get_logger("CarlikeBotSystemHardware"), "%.1f seconds left...", hw_start_sec_ - i);
+  }
+
+  for (uint i = 0; i < hw_positions_.size(); i++)
+  {
+    hw_positions_[i] = 0.0;
+    hw_velocities_[i] = 0.0;
+    hw_commands_[i] = 0.0;
+  }
+
+  RCLCPP_INFO(rclcpp::get_logger("CarlikeBotSystemHardware"), "Successfully activated!");
+
+  return hardware_interface::CallbackReturn::SUCCESS;
+}
+
+hardware_interface::CallbackReturn CarlikeBotSystemHardware::on_deactivate(
+  const rclcpp_lifecycle::State & /*previous_state*/)
+{
+  // BEGIN: This part here is for exemplary purposes - Please do not copy to your production code
+  RCLCPP_INFO(rclcpp::get_logger("CarlikeBotSystemHardware"), "Deactivating ...please wait...");
+
+  for (auto i = 0; i < hw_stop_sec_; i++)
+  {
+    rclcpp::sleep_for(std::chrono::seconds(1));
+    RCLCPP_INFO(
+      rclcpp::get_logger("CarlikeBotSystemHardware"), "%.1f seconds left...", hw_stop_sec_ - i);
+  }
+  // END: This part here is for exemplary purposes - Please do not copy to your production code
+  RCLCPP_INFO(rclcpp::get_logger("CarlikeBotSystemHardware"), "Successfully deactivated!");
+
+  return hardware_interface::CallbackReturn::SUCCESS;
+}
+
+hardware_interface::return_type CarlikeBotSystemHardware::read(
+  const rclcpp::Time & /*time*/, const rclcpp::Duration & period)
+{
+  // BEGIN: This part here is for exemplary purposes - Please do not copy to your production code
+  for (std::size_t i = 0; i < hw_velocities_.size(); ++i)
+  {
+    // Simulate DiffBot wheels's movement as a first-order system
+    // Update the joint status: this is a revolute joint without any limit.
+    // Simply integrates
+    hw_positions_[i] += hw_velocities_[i] * period.seconds();
+
+    RCLCPP_INFO(
+      rclcpp::get_logger("CarlikeBotSystemHardware"),
+      "Got position state %.5f and velocity state %.5f for '%s'", hw_positions_[i],
+      hw_velocities_[i], info_.joints[i].name.c_str());
+  }
+  // END: This part here is for exemplary purposes - Please do not copy to your production code
+
+  return hardware_interface::return_type::OK;
+}
+
+hardware_interface::return_type ros2_control_demo_example_11 ::CarlikeBotSystemHardware::write(
+  const rclcpp::Time & /*time*/, const rclcpp::Duration & /*period*/)
+{
+  RCLCPP_INFO(rclcpp::get_logger("CarlikeBotSystemHardware"), "Writing...");
+
+  for (auto i = 0u; i < hw_commands_.size(); ++i)
+  {
+    hw_velocities_[i] = hw_commands_[i];
+
+    RCLCPP_INFO(
+      rclcpp::get_logger("CarlikeBotSystemHardware"), "Got command %.5f for '%s'", hw_commands_[i],
+      info_.joints[i].name.c_str());
+  }
```

Comment:
I think the problem comes from this part of the code: steering command is position (angle), and not a velocity.